app/src/main/java/com/bogdan/codeforceswatcher/features/auth/SignUpComposeActivity.kt

```diff
@@ -0,0 +1,298 @@
+package com.bogdan.codeforceswatcher.features.auth
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.runtime.*
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusDirection
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.ParagraphStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.MutableLiveData
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.components.WebViewActivity
+import com.bogdan.codeforceswatcher.components.compose.*
+import com.bogdan.codeforceswatcher.components.compose.theme.AlgoismeTheme
+import io.xorum.codeforceswatcher.features.auth.redux.AuthRequests
+import io.xorum.codeforceswatcher.features.auth.redux.AuthState
+import io.xorum.codeforceswatcher.redux.store
+import io.xorum.codeforceswatcher.util.Constants.PRIVACY_POLICY_LINK
+import io.xorum.codeforceswatcher.util.Constants.TERMS_AND_CONDITIONS_LINK
+import tw.geothings.rekotlin.StoreSubscriber
+
+class SignUpComposeActivity : ComponentActivity(), StoreSubscriber<AuthState> {
+    @ExperimentalComposeUiApi
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            AlgoismeTheme {
+                SignUpScreen()
+            }
+        }
+    }
+
+    private val authState = MutableLiveData<AuthState>()
+
+    private var email = ""
+    private var password = ""
+    private var confirmedPassword = ""
+
+    @ExperimentalComposeUiApi
+    @Composable
+    private fun SignUpScreen() {
+        val localFocusManager = LocalFocusManager.current
+
+        var isPrivacyPolicyAccepted by remember { mutableStateOf(false) }
+        val authState by authState.observeAsState()
+
+        Scaffold(
+            modifier = Modifier.fillMaxSize(),
+            topBar = {
```

Comment:
WDYT about extracting private composables `TopBar`, `BottomBar` and `Content`? This will make separate methods smaller and easier to read.

Reply:
Good idea👍 I think we should do this in the refactoring task.

Reply:
OK, will you create one?

Reply:
It has already been created.